src/parser/mod.rs

```diff
@@ -4139,6 +4160,163 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub fn parse_create_trigger(&mut self, or_replace: bool) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
+            let name = self.parse_object_name(false)?;
+            let period = self.parse_trigger_period()?;
+
+            let event = self.parse_keyword_separated(Keyword::OR, Parser::parse_trigger_event)?;
+            let table_name = if self.parse_keyword(Keyword::ON) {
+                self.parse_object_name(false)?
+            } else {
+                return self.expected("keyword `ON`", self.peek_token());
+            };
+
+            let mut referencing = vec![];
+            if self.parse_keyword(Keyword::REFERENCING) {
+                while let Some(refer) = self.parse_trigger_referencing()? {
+                    referencing.push(refer);
+                }
+            }
+
+            let for_each = if self.parse_keywords(&[Keyword::FOR]) {
+                let _ = self.parse_keyword(Keyword::EACH);
```

Comment:
It'll change `FOR` to `FOR EACH` while formatting SQL, perhaps we should record if `EACH` keyword existing?

Reply:
Ok, doing.

Reply:
Done, and included both "FOR" and "FOR EACH" in the test suite.